spec/browsers/downcast_user_agent_spec.rb

```diff
@@ -0,0 +1,66 @@
+require 'user_agent'
+
+shared_examples_for 'Downcast' do |version, platform, os, type|
+  it "returns 'Downcast' as its browser" do
+    expect(useragent.browser).to eq('Downcast')
+  end
+
+  it "returns '#{version}' as its version" do
+    expect(useragent.version).to eq(version)
+  end
+
+  it "returns '#{platform}' as its platform" do
+    expect(useragent.platform).to eq(platform)
+  end
+
+  it "returns '#{os}' as its operating system" do
+    expect(useragent.os).to eq(os)
+  end
+
+  if type == :mobile
+    it { expect(useragent).to be_mobile }
+    it { expect(useragent).not_to be_desktop }
+  else
```

Comment:
Either make it a boolean or handle :desktop specifically to avoid unhandled extra states